presto-docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -112,3 +112,38 @@ To enable SSL/TLS for Presto internal communication, do the following:
 
         internal-communication.https.keystore.path=<keystore path>
         internal-communication.https.keystore.key=<keystore password>
+
+
+Enabling encryption introduces performance impact. Performance degradation can
+vary based on the environment, queries, concurrency.
+
+For the queries, that are not require to much data to be transferred between the Presto
+nodes (such as simple `SELECT count(*) FROM table`), performance
+impact is unnoticeable.
+
+However for the queries that are CPU intensive, and require considerable amount of data
+to be transferred between the nodes (for example when repartitioning is needed:
+JOIN query with distributed join enabled, or queries with GROUP BY), performance impact
+might be considerable. Slowdown may vary from 10% to even 100%+ depending on the network
+traffic and the CPU utilization.
+
+TLS encryption uses it as a source of random numbers the `/dev/urandom` system device,
+which is system-wide synchronized, and has a limited throughput.
+On the environments with high network bandwidth (e.g. InfiniBand), powerful CPU
+(multi-core high-end server CPU's) and high concurrency, it might become a bottleneck.
```

Comment:
"might become a bottleneck" -> I would argue that such environments (fast network, lots of CPU cores, and high concurrency) are likely to be typical for Presto production deployments.
Please consider making the default settings appropriate for those cases.